text_handler.py

```diff
@@ -26,42 +29,58 @@ async def __handleYoutubeDownload(self, message: types.Message):
         if not str(message.chat.id) in config.WHITELIST_CHAT_ID:
             return
         try:
-            yt = YouTube(message.text)
-            author = yt.author
-            length = yt.length
-            title = yt.title
-            stream = (yt.streams
-                      .filter(only_audio=True, file_extension='mp4')
-                      .order_by('bitrate').desc().first())
-            filesize = stream.filesize
-            video_id = extract.video_id(message.text)
-            video_url = "youtu.be/" + video_id
+            ydl_opts = {
+                    'writethumbnail': True,
+                    'outtmpl': 'downloads/%(extractor)s_%(id)s.%(ext)s',
+                    'format': 'bestaudio',
+            }
+            with YoutubeDL(ydl_opts) as ydl:
```

Comment:
we should in future move all this logic into a separate class, i had this idea for a long time, but was too lazy to make it happen. The idea is to make a class, for example call it **AudioDownloader** and here we would just need to call **AudioDownloader.download(url)**, when you have time and desire you can check how to make classes and work with them

Reply:
would that class in a separate file? or inside text_handler.py?

Reply:
in a separate file

Reply:
Pushed new commit and did this.